rpm/gratia-probe.spec

```diff
@@ -258,19 +258,21 @@ git_commit_id=$(gzip -d < %{SOURCE0} | git get-tar-commit-id)
   install -d $RPM_BUILD_ROOT/%{_sysconfdir}/condor/config.d
   install -m 644 condor/99_gratia.conf $RPM_BUILD_ROOT/%{_sysconfdir}/condor/config.d/99_gratia.conf
   install -m 644 condor/99_gratia-gwms.conf $RPM_BUILD_ROOT/%{_sysconfdir}/condor/config.d/99_gratia-gwms.conf
+  rm $RPM_BUILD_ROOT%{_datadir}/gratia/condor/99_gratia.conf
   rm $RPM_BUILD_ROOT%{_datadir}/gratia/condor/99_gratia-gwms.conf
 
   # Install the htcondor-ce configuration
   install -d $RPM_BUILD_ROOT/%{_sysconfdir}/condor-ce/config.d
-  install -m 644 condor/99_gratia.conf $RPM_BUILD_ROOT/%{_sysconfdir}/condor-ce/config.d/99_gratia.conf
-  rm $RPM_BUILD_ROOT%{_datadir}/gratia/condor/99_gratia.conf
-  install -m 644 condor/gratia-probe-htcondor-ce.cron $RPM_BUILD_ROOT/%{_sysconfdir}/cron.d/gratia-probe-htcondor-ce.cron
+  install -m 644 condor/99_gratia-ce.conf $RPM_BUILD_ROOT/%{_sysconfdir}/condor-ce/config.d/99_gratia-ce.conf
   install -d $RPM_BUILD_ROOT%{_datadir}/gratia/htcondor-ce/
   install -m 755 condor/condor_meter $RPM_BUILD_ROOT%{_datadir}/gratia/htcondor-ce/
   # Copy the condor configuration
   install -d $RPM_BUILD_ROOT/%{_sysconfdir}/gratia/htcondor-ce
   install -m 644 $RPM_BUILD_ROOT/%{_sysconfdir}/gratia/condor/ProbeConfig $RPM_BUILD_ROOT/%{_sysconfdir}/gratia/htcondor-ce/ProbeConfig
   sed -i 's/ProbeName="condor:/ProbeName="htcondor-ce:/' $RPM_BUILD_ROOT/%{_sysconfdir}/gratia/htcondor-ce/ProbeConfig
+  sed -i '/WorkingFolder="/s/"[^"]*"/"/var/lib/condor-ce/"' $RPM_BUILD_ROOT/%{_sysconfdir}/gratia/htcondor-ce/ProbeConfig
+  sed -i '/LogFolder="/s/"[^"]*"/"/var/log/condor-ce/"' $RPM_BUILD_ROOT/%{_sysconfdir}/gratia/htcondor-ce/ProbeConfig
+  sed -i '/LogFolder="/a\    Lockfile="/var/lock/condor-ce/gratia.lock"' $RPM_BUILD_ROOT/%{_sysconfdir}/gratia/htcondor-ce/ProbeConfig
```

Comment:
This is how we set the default values in the ProbeConfigs???

How does Grata know to look for history files in "/var/lib/condor-ce/gratia_data"?

Reply:
>This is how we set the default values in the ProbeConfigs????????????????????????????

There are two mechanisms for customization that i know of; putting stuff in a `ProbeConfig.add` in the probe subdir, or doing a `sed` in the rpm spec to replace `@PROBE_SPECIFIC_DATA@`.  But these both _add_ stuff rather than replace values.

Otherwise, I don't think we have a standard way to set the values in the ProbeConfig.

I agree (with your "???") that it looks pretty ugly to do it with sed like this.  It would be cleaner to do something like the configure-nonroot-gratia.py script does, which is to say, parse the xml, update it, and dump the xml again.  But then you lose the nice one-item-per-line layout of the original that makes editing the ProbeConfig by hand suck a lot less.

I could make a function for updating ProbeConfig values, which would at least make this part look less gross.

>How does Grata know to look for history files in "/var/lib/condor-ce/gratia_data"?

Does gratia not use the `WorkingFolder` setting for this?

There is also a `DataFolder="/var/lib/gratia/data/"` setting in the default ProbeConfig, but your configure-nonroot-gratia.py script does not touch that one.

Reply:
> Does gratia not use the WorkingFolder setting for this?

I don't think so. This is where a Hosted CE is dumping history files:

```
# condor_ce_config_val -v PER_JOB_HISTORY_DIR
PER_JOB_HISTORY_DIR = /var/lib/gratia/data
 # at: /etc/condor-ce/config.d/99_gratia.conf, line 5
 # raw: PER_JOB_HISTORY_DIR = /var/lib/gratia/data
```

And it has the following `*Folder` attributes set in the ProbeConfig:

```
CondorCEHistoryFolder="/var/lib/gratia/condorce_data"
DataFolder="/var/lib/gratia/data/"
WorkingFolder="/var/lib/condor-ce/"
```

There appears to be a `/var/lib/condor-ce/gratiafiles/subdir.<munged ProbeName>/outbox` with nothing in it. The Hosted CE setup is broken, though: history files are written to `PER_JOB_HISTORY_DIR` but they are not being removed, either because of a permission issue (`condor_meter` is run by the `condor` user as a schedd cron) or because of the incorrect `*Folder` configurations. What are each of these folders for?

Reply:
Ok, i see in the jira ticket the path `/var/lib/condor-ce/gratia_data` was mentioned (which i then used) where probably `/var/lib/condor-ce/gratia/data` was intended (which in the other comments is what `PER_JOB_HISTORY_DIR` was set to, as well as the default value of `DataFolder`.  I'm updating this which hopefully is what's needed.